File: node/application/usecases/CreateUser.ts
import ID from "../../domain/user/ID";
import User from "../../domain/user/User";
import { UserRepository } from "../ports/UserRepository";

export class CreateUser{
    constructor(public userRepo: UserRepository){
        this.userRepo = userRepo
    }
    async execute({
        id,
        phoneNumber,
        role,
        age,
        sex
    }){
        const user = new User({
            id: new ID(),
            phoneNumber: phoneNumber,
            role: role,
            age: age,
            sex: sex
        })
        await this.userRepo.save(user)
        return user
    }
}

File: python/application/usecases/create_user.py
from python.application.ports.user_repository import UserRepository
from python.domain.entities.user import User
from python.domain.value_objects.id import ID


class CreateUser:
    def __init__(self, user_repo: UserRepository) -> None:
        self.user_repo = user_repo
    
    async def execute(self, id, phone_number, role, age, sex):
        user = User(uid=ID(), phone_number=phone_number, role=role, age=age, sex=sex)
        await self.user_repo.save(user)
        return user